odbc/src/api/functions.rs

```diff
@@ -1207,7 +1276,41 @@ unsafe fn sql_fetch_helper(statement_handle: HStmt, function_name: &str) -> SqlR
 
         *stmt.var_data_cache.write().unwrap() = Some(HashMap::new());
 
-        if !warnings_opt.is_empty() {
+        let mut success_with_info_encountered = false;
+
+        if let Some(ref bound_cols) = stmt.bound_cols.read().unwrap().as_ref() {
```

Comment:
also a small nit but maybe a comment at the top of this block would be helpful -- jsut noting we copy the data from the result set into the bound buffers if we have bound columns